esp-hal/build.rs

```diff
@@ -13,6 +13,23 @@ use esp_config::{generate_config, Validator, Value};
 use esp_metadata::{Chip, Config};
 
 fn main() -> Result<(), Box<dyn Error>> {
+    if ![
+        "xtensa-esp32-none-elf",
+        "xtensa-esp32s2-none-elf",
+        "xtensa-esp32s3-none-elf",
+        "riscv32imc-unknown-none-elf",
+        "riscv32imac-unknown-none-elf",
+    ]
+    .contains(&std::env::var("TARGET").unwrap_or_default().as_str())
```

Comment:
Since we know the chip, we might as well check if the target matches what the chip is, but this is definitely better than nothing.

Reply:
Also a bit nitpicky, but perhaps just so that we don't need to change this in the future we can do something like:

```rust
let targets = Chip::iter().map(|chip| chip.target()).collect::<HashSet<_>>();
```